cmds/core/shasum/shasum.go

```diff
@@ -5,83 +5,100 @@
 package main
 
 import (
+	"bufio"
 	"crypto/sha1"
 	"crypto/sha256"
 	"fmt"
 	"io"
+	"log"
 	"os"
 
 	"github.com/spf13/pflag"
 )
 
-func helpPrinter() {
-	fmt.Printf("Usage:\nshasum -a <algorithm> <File Name>\n")
-	pflag.PrintDefaults()
-	os.Exit(0)
-}
+var (
+	algorithm = pflag.IntP("algorithmhttps://github.com/u-root/u-root/pull/2319", "a", 1, "SHA algorithm, valid args are 1 and 256")
+	help      = pflag.BoolP("help", "h", false, "Show this help and exit")
+)
 
-func versionPrinter() {
-	fmt.Println("shasum utility, URoot Version.")
-	os.Exit(0)
-}
+var usage = "Usage:\nshasum -a <algorithm> <File Name>"
 
-func getInput(fileName string) (input []byte, err error) {
-	if fileName != "" {
-		return os.ReadFile(fileName)
-	}
-	return io.ReadAll(os.Stdin)
+func helpPrinter() {
+	fmt.Println(usage)
+	pflag.PrintDefaults()
 }
 
 //
 // shaPrinter prints sha1/sha256 of given data. The
 // value of algorithm is expected to be 1 for SHA1
 // and 256 for SHA256
 //
-func shaPrinter(algorithm int, data []byte) string {
-	var sha string
-	if algorithm == 256 {
-		sha = fmt.Sprintf("%x", sha256.Sum256(data))
-	} else if algorithm == 1 {
-		sha = fmt.Sprintf("%x", sha1.Sum(data))
+func shaPrinter(w io.Writer, data []byte) error {
+	if *algorithm == 256 {
+		fmt.Fprintf(w, "%x", sha256.Sum256(data))
+	} else if *algorithm == 1 {
+		fmt.Fprintf(w, "%x", sha1.Sum(data))
 	} else {
-		fmt.Fprintf(os.Stderr, "Invalid algorithm")
-		return ""
+		return fmt.Errorf("invalid algorithm")
 	}
-	return sha
+	return nil
 }
 
-func main() {
-	var (
-		algorithm int
-		help      bool
-		version   bool
-	)
-	cliArgs := ""
-	pflag.IntVarP(&algorithm, "algorithm", "a", 1, "SHA algorithm, valid args are 1 and 256")
-	pflag.BoolVarP(&help, "help", "h", false, "Show this help and exit")
-	pflag.BoolVarP(&version, "version", "v", false, "Print Version")
-	pflag.Parse()
+func getInput(filename string) ([]byte, error) {
+	file, err := os.Open(filename)
 
-	if help {
-		helpPrinter()
+	if err != nil {
+		return nil, err
 	}
+	defer file.Close()
 
-	if version {
-		versionPrinter()
-	}
-	if len(pflag.Args()) == 1 {
-		cliArgs = pflag.Args()[0]
-	}
-	input, err := getInput(cliArgs)
+	stats, err := file.Stat()
 	if err != nil {
-		fmt.Println("Error getting input.")
-		os.Exit(-1)
+		return nil, err
 	}
-	fmt.Printf("%s ", shaPrinter(algorithm, input))
-	if cliArgs == "" {
-		fmt.Printf(" -\n")
+
+	size := stats.Size()
+	bytes := make([]byte, size)
+
+	buf := bufio.NewReader(file)
+	_, err = buf.Read(bytes)
```

Comment:
Do we need buffered IO capabilities, if reading a file?

Reply:
Don't think so, thanks for reporting!